routingapi/api.go

```diff
@@ -0,0 +1,145 @@
+package routingapi
+
+import (
+	"fmt"
+	"time"
+
+	"code.cloudfoundry.org/route-registrar/config"
+
+	uaaclient "code.cloudfoundry.org/uaa-go-client"
+	uaaconfig "code.cloudfoundry.org/uaa-go-client/config"
+
+	"code.cloudfoundry.org/routing-api/models"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/routing-api"
+)
+
+type RoutingAPI interface {
+	Init(api_config config.RoutingAPI) error
+	RegisterRoute(route config.Route) error
+	UnregisterRoute(route config.Route) error
+	Close()
+}
+
+type apiState struct {
+	logger          lager.Logger
+	uaaClient       uaaclient.Client
+	apiClient       routing_api.Client
```

Comment:
please rename this to `routingAPI` - go does not have underscores for package names (I know the package is misnamed, haven't gotten around to renaming it).